diag_manager2/fms_diag_object.F90

```diff
@@ -270,29 +299,46 @@ subroutine diag_ob_registered(objin , reg)
  logical                      , intent(in)   :: reg !< If registering, this is true
  objin%registered = reg
 end subroutine diag_ob_registered
+
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-!> \brief Copies the calling object into the object that is the argument of the subroutine
+!> \brief Copies the calling object into the object that is the argument of the subroutine.
+!! The input object is not copied if it is not registered.
 subroutine copy_diag_obj(objin , objout)
- class (fms_diag_object)      , intent(in)                :: objin
- class (fms_diag_object)      , intent(inout) , allocatable :: objout !< The destination of the copy
-select type (objout)
- class is (fms_diag_object)
+ class (fms_diag_object)    , intent(in)                :: objin
+ class (fms_diag_object)    , intent(inout) , allocatable :: objout !< The destination of the copy
 
-  if (allocated(objin%registered)) then
-     objout%registered = objin%registered
-  else
-     call diag_error("copy_diag_obj", "You can only copy objects that have been registered",warning)
-  endif
-!     type (diag_fields_type)                           :: diag_field         !< info from diag_table
-!     type (diag_files_type),allocatable, dimension(:)  :: diag_file          !< info from diag_table
+ select type (objout)
+    class is (fms_diag_object)
+    if (allocated(objin%registered)) then
+        objout%registered = objin%registered
+    else
+        call diag_error("copy_diag_obj", "You can only copy objects that have been registered",warning)
+    endif
```

Comment:
In your comments for the description of the routine, you have 

> !! The input object is not copied if it is not registered.

If this is true, then the routine should return here and not continue.